lib/timescaledb/model.ex

```diff
@@ -0,0 +1,126 @@
+defmodule Membrane.RTC.Engine.TimescaleDB.Model do
+  @moduledoc """
+  Module responsible for putting data to TimescaleDB.
+  """
+
+  import Ecto.Query
+
+  alias Membrane.RTC.Engine.TimescaleDB.Repo
+
+  alias Membrane.RTC.Engine.TimescaleDB.Model.{
+    PeerMetrics,
+    PeerToRoom,
+    TrackMetrics,
+    TrackToPeer
+  }
+
+  @doc """
+  Takes `Membrane.RTC.Engine.Metrics.rtc_engine_report()` and puts it to database.
+  """
+  @spec insert_report(Membrane.RTC.Engine.Metrics.rtc_engine_report()) :: :ok
+  def insert_report(report) do
+    for {{:room_id, room_id}, room_report} <- report do
+      for {{:peer_id, peer_id}, peer_report} <- room_report do
+        {peer_metrics, tracks_reports} =
+          Enum.split_with(peer_report, fn {key, _value} -> is_atom(key) end)
+
+        %{peer_id: peer_id, room_id: room_id, time: NaiveDateTime.utc_now()}
+        |> upsert_peer_to_room()
+
+        Map.new([peer_id: peer_id, time: NaiveDateTime.utc_now()] ++ peer_metrics)
+        |> update_if_exists(:"peer.metadata", &inspect/1)
+        |> insert_peer_metrics()
+
+        for {{:track_id, track_id}, track_report} <- tracks_reports do
+          %{track_id: track_id, peer_id: peer_id, time: NaiveDateTime.utc_now()}
+          |> upsert_track_to_peer()
+
+          Map.merge(track_report, %{track_id: track_id, time: NaiveDateTime.utc_now()})
```

Comment:
1. This time insertions also belong to the changeset.
2. If all they describe is insertion time, you should rather add [`timestamps`](https://hexdocs.pm/ecto/Ecto.Schema.html#timestamps/1) to the schema, which will enable 
3. You should use DateTIme (and `:utc_datetime` ecto field type). You may find this useful: http://www.creativedeletion.com/2019/06/17/utc-timestamps-in-ecto.html
4. `:naive_datetime` and `:utc_datetime` have a precision of seconds. If that's not enough you should use `:utc_datetime_usec`/`:naive_datetime_usec` - the Elixir type will stay the same, but now it's trimmed when storing in DB